### DISCUSSION TITLE
Reverse Traversal [feature request]

### DISCUSSION THREAD

**Discussion by JohanMeppelinkDRA** [networkit/networkit#993] [Q&A]
With a directed graph we can use traversal.BFSfrom (for example) to find all nodes downstream.
Would it be possible to reverse the search and find all nodes upstream?

Simple path:
A -> B -> C

From A we get, {A B C}
But sometimes we want to now what nodes are leading to C {C, B,. A}.

Could this be added as a feature? Perhaps this is already possible, but I missed it somehow.

ReverseBFS appeared like it would do the trick, but it seems to require an end node before returing the nodes in between.

**Comment by fabratu** [comment]
For BFS (and all SSSP algorithms) you can also store paths / predecessors for each graph. So in principal you can traverse backwards. Also you can extract any path in reverse order (`getPath(u, forward=False)`). But I am not sure that covers your use case.
